k9mail/src/main/java/com/fsck/k9/preferences/DeletePolicy.java

```diff
@@ -0,0 +1,34 @@
+package com.fsck.k9.preferences;
+
+import android.util.Log;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.K9;
+import com.fsck.k9.mail.ServerSettings;
+import com.fsck.k9.mail.store.imap.ImapStore;
+import com.fsck.k9.mail.store.pop3.Pop3Store;
+import com.fsck.k9.mail.store.webdav.WebDavStore;
+
+/**
+ * Defaults for mail deletion.
+ */
+public class DeletePolicy {
```

Comment:
What I would like is for Account.DELETE_POLICY_* to be an enum, which we could reuse.

However it would be inconsistent with the rest of the constants (TYPE_*, EXPUNGE_*) in that class - I could do all of them as part of this refactor, or just do DELETE_POLICY_* ?

Not quite sure about coupling deletion policy to server settings, they seem unrelated to me. If there was an enum encompassing {WebDavStore,Pop3Store,ImapStore}.STORE_TYPE I would be happy to put it in there.